Abyss.Web/ClientApp/src/app/online/online.component.ts

```diff
@@ -26,6 +27,21 @@ export class OnlineComponent implements OnInit, OnDestroy {
                 this.clients = clients;
             });
         });
+        this.hub.onclose(() => {
+            this.ngZone.run(() => {
+                this.hubReady = false;
```

Comment:
Do you actually need to subscribe to these events? Seems like a bit of a waste when on the ngOnDestroy you can just call `this.hub.stop()`. Does it matter if it's ready?

Reply:
Probably don't need the onclose to be fair as I think that is only called when manually closing it rather than when connecting is lost

Reply:
The hubReady is used to block the UI until connected as it's tightly connected atm and won't work without, but as you said in your other comment it shouldn't be tightly connected anyway